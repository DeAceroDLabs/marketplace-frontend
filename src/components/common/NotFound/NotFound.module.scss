@use 'styles/colors';
@use 'styles/typography';

.container {
    margin: 5%;
    height: 544px;
    display: flex;
    justify-content: center;
    align-items: center;
    box-shadow: 0px 5px 20px rgba(181, 181, 181, 0.25);

    .image {
        width: 330px;
    }

    .containerInfo {
        width: 40%;
        margin: 3%;

        .title {

            font-weight: typography.$header-s-weight;
            font-size: typography.$header-m-size;
        }

        .info {
            font-size: typography.$body-m-size;
        }
    }
}
