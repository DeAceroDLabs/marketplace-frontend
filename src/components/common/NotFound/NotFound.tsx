import styles from "./NotFound.module.scss";

interface NotFoundProps {
  title:  string;
  imgSrc: string;
  children: React.ReactNode;
  
}

const NotFound: React.FC<NotFoundProps> = ({
  title,
  imgSrc,
  children,
}) => {
  return (
    <div >
      <div className={styles.container}>
        <img className={styles.image} src={imgSrc} alt="Not found img" />
        <div className={styles.containerInfo}>
          <div className={styles.title}>{title}</div>
          <div className={styles.info}>{children}</div>
        </div>
      </div>
    </div>
  );
};

export default NotFound;
