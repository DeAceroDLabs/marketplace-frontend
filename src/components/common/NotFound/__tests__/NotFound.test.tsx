import renderer from "react-test-renderer";
import NotFound from "../NotFound";

const NotFoundPropsMock = {
  title: "Mock Title",
  imgSrc: "path/to/mock/image",
};

it("renders NotFound with no issue", () => {
  const component = renderer.create(
    <NotFound {...NotFoundPropsMock}>
      <></>
    </NotFound>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
