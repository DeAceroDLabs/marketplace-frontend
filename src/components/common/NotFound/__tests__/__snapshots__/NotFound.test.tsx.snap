// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders NotFound with no issue 1`] = `
<div>
  <div
    className="container"
  >
    <img
      alt="Not found img"
      className="image"
      src="path/to/mock/image"
    />
    <div
      className="containerInfo"
    >
      <div
        className="title"
      >
        Mock Title
      </div>
      <div
        className="info"
      />
    </div>
  </div>
</div>
`;
